packages/apps/shopify-api/runtime/http/cookies.ts

```diff
@@ -175,25 +175,40 @@ export class Cookies {
     this.updateHeader();
   }
 
+  private cookieExists(cookieName: string) {
+    return !!this.get(cookieName);
+  }
+
+  private deleteInvalidCookies(...cookieNames: string[]): void {
+    cookieNames.forEach((cookieName) => this.deleteCookie(cookieName));
+  }
+
   async isSignedCookieValid(cookieName: string): Promise<boolean> {
     const signedCookieName = `${cookieName}.sig`;
-    // No cookie or no signature cookie makes the cookie it invalid.
-    if (!this.get(cookieName) || !this.get(signedCookieName)) {
-      this.deleteCookie(signedCookieName);
-      this.deleteCookie(cookieName);
+    if (
+      !this.cookieExists(cookieName) ||
+      !this.cookieExists(signedCookieName)
+    ) {
+      this.deleteInvalidCookies(cookieName, signedCookieName);
+      return false;
+    }
+    const cookieValue = this.get(cookieName);
```

Comment:
Are lines 195-199 duplicating lines 188-194?
I am not sure I am understanding why both are necessary?